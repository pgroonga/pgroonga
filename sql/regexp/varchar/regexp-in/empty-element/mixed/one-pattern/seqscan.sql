CREATE TABLE memos (
  content varchar(256)
);

INSERT INTO memos VALUES ('Groonga is fast full text search engine');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga');

SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;

EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE content &~| ARRAY['', '\Agroonga']::varchar[];

SELECT *
  FROM memos
 WHERE content &~| ARRAY['', '\Agroonga']::varchar[];

DROP TABLE memos;
