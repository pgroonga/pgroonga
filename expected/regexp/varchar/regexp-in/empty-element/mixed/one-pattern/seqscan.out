CREATE TABLE memos (
  content varchar(256)
);
INSERT INTO memos VALUES ('Groonga is fast full text search engine');
INSERT INTO memos VALUES ('PGroonga is a PostgreSQL extension that uses Groonga');
SET enable_seqscan = on;
SET enable_indexscan = off;
SET enable_bitmapscan = off;
EXPLAIN (COSTS OFF)
SELECT *
  FROM memos
 WHERE content &~| ARRAY['', '\Agroonga']::varchar[];
                            QUERY PLAN                            
------------------------------------------------------------------
 Seq Scan on memos
   Filter: (content &~| '{"","\\Agroonga"}'::character varying[])
(2 rows)

SELECT *
  FROM memos
 WHERE content &~| ARRAY['', '\Agroonga']::varchar[];
                 content                 
-----------------------------------------
 Groonga is fast full text search engine
(1 row)

DROP TABLE memos;
